src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -4689,10 +4689,14 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
             step = llvm::ConstantInt::get(context,
                 llvm::APInt(32, 0));
         }
-        bool flag = str->getType()->getContainedType(0)->isPointerTy();
+        #if LLVM_VERSION_MAJOR >= 16
```

Comment:
```suggestion
        #if LLVM_VERSION_MAJOR >= 17
```
I believe this should be right version

Reply:
Hey Pranav, thanks for the reviews. After discussing with Thirumalai I think I have the correct changes now !